client/trino-jdbc/src/main/java/io/trino/jdbc/NonRegisteringTrinoDriver.java

```diff
@@ -55,7 +57,7 @@ public Connection connect(String url, Properties info)
         OkHttpClient.Builder builder = httpClient.newBuilder();
         uri.setupClient(builder);
 
-        return new TrinoConnection(uri, builder.build());
+        return new TrinoConnection(uri, OkHttpTelemetry.builder(GlobalOpenTelemetry.get()).build().newCallFactory(builder.build()));
```

Comment:
Note, when OpenTelemetry is not configured, a `noop` implementation is returned by `GlobalOpenTelemetry.get()`.